tests/tests_data_model/test_models/test_conceptual/test_property.py

```diff
@@ -0,0 +1,126 @@
+from datetime import date
+
+import pytest
+from pydantic import ValidationError
+
+from cognite.neat._data_model.models.conceptual._properties import Property
+from cognite.neat._data_model.models.entities import URI, ConceptEntity, String, UnknownEntity
+from cognite.neat._data_model.models.entities._data_types import AnyURI, Boolean, Date, Float, Integer
+
+
+class TestProperty:
+    def test_property_creation_with_valid_data(self):
+        """Test creating a Property with valid data."""
+        property_obj = Property(
+            value_type=String(),
+            min_count=1,
+            max_count=5,
+        )
+
+        assert isinstance(property_obj.value_type, String)
+        assert property_obj.min_count == 1
+        assert property_obj.max_count == 5
+
+    def test_property_with_concept_entity_value_type(self):
+        """Test creating a Property with ConceptEntity as value type."""
+        concept_entity = ConceptEntity(prefix="ex", suffix="concept_1")
+        property_obj = Property(value_type=concept_entity)
+
+        assert property_obj.value_type == concept_entity
+
+    def test_property_with_unknown_entity_value_type(self):
+        """Test creating a Property with UnknownEntity as value type."""
+        property_obj = Property(value_type=UnknownEntity())
+
+        assert isinstance(property_obj.value_type, UnknownEntity)
```

Comment:
I am skeptical about these types of tests. They are essentially testing `pydantic` functionality, which I do not think we have to do as that increases the maintenance cost of this code with no benefit.

Instead, I suggest focusing on the error cases were you have custom validators. In addition, how to translate the pydantic errors into errors for our users - see https://github.com/cognitedata/neat/pull/1286